### PR TITLE
Finishing primitives tests (finally)

### PR DESCRIPTION
**Pull request by elevin72**

### CODE REVIEW

File: src/test/primitives/PointTests.java [hunk 1]
```diff
@@ -1,59 +1,121 @@
...
 	}
 
 	/**
 	 * Test Point.equals
 	 */
 	@Test
 	public void equals() {
-		// TODO: implement
+        Assert.assertEquals("Equal points are not considered equal.", p, new Point(1,2,3));
+        Assert.assertNotEquals("Equal points are not considered equal.", p, new Point(2,3,4));
```
Comment (abrahammurciano) [inline src/test/primitives/PointTests.java:119]:
```suggestion
        Assert.assertNotEquals("Non-equal points are considered equal.", p, new Point(2,3,4));
```

**Review by abrahammurciano — CHANGES_REQUESTED** [review]
use tabs and restore the docs

**Review by abrahammurciano — CHANGES_REQUESTED** [review]
where are my tabs?

**Review by abrahammurciano — APPROVED** [review]

File: src/test/primitives/PointTests.java [hunk 2]
```diff
@@ -1,59 +1,121 @@
...
 
+import util.DoubleCompare;
+
 /**
  * Test the functions of the point class
  *
  * @author Abraham Murciano
  * @author Eli Levin
  */
 public class PointTests {
```
Comment (abrahammurciano) [inline src/test/primitives/PointTests.java:14]: Can you add a short comment for each test

File: src/test/geometries/CylinderTests.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		// test on round side
+		// 30.973500981126143 23, 12.639748528310783
+		NormalizedVector calc = base.normal(new Point(30.973500981126143, 23, 13.639748528310783));
+		NormalizedVector actual = new NormalizedVector(4,0,-6);
+		Assert.assertEquals("Normalized vectors should be equal", calc, actual);
+
+		// test on flat side
+		calc =	base.normal(new Point(7.79639313525896, 11.780511727620528, 3.517181606066203));
+		actual = new NormalizedVector(6,5,4);
+		Assert.assertEquals("Normalized vectors should be equal", calc, actual);
```
[the remark this answers is not in the document]
Comment (elevin72) [reply src/test/geometries/CylinderTests.java:27]: No, it should always be outward facing from the shape, if the shape has an in/out (not like a plane).

File: src/test/geometries/CylinderTests.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import primitives.Point;
+import primitives.Ray;
+import primitives.Vector;
+
+public class CylinderTests {
+	public final Ray ray = new Ray(new Point(3,2,1), new Vector(6,5,4));
+	public final Cylinder base = new Cylinder(ray, 5, 10);
+
+	@Test 
+	public void normal() {
```
[the remark this answers is not in the document]
Comment (elevin72) [reply src/test/geometries/CylinderTests.java:16]: 1) Like on the corner where the flat part meets the curve?
2) Why could we expect a difference at the center of the disk?

File: src/test/geometries/SphereTests.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+	@Test
+	public void normal() {
+		NormalizedVector calc, actual;
+
+		// Test at direction (2,1,1)
+		calc =
+			s.normal(new Point(5.08248290463863016366, 4.04124145231931508183, 5.04124145231931508183));
+		actual = new NormalizedVector(2, 1, 1);
+		Assert.assertEquals("Normalized vectors to the sphere should be equal.", calc, actual);
```
[the remark this answers is not in the document]
Comment (elevin72) [reply src/test/geometries/SphereTests.java:26]: see above

File: src/test/geometries/CylinderTests.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (abrahammurciano) [reply src/test/geometries/CylinderTests.java:16]: 1. Yes
2. Idk it's an edge case. We're not supposed to take the implementation into account. Tho I guess it's the same as any other point on the disk, so idk. Do what u think for this one

[the remark this answers is not in the document]
Comment (elevin72) [reply src/test/geometries/CylinderTests.java:16]: 1. I should just check that the normal is either correct wrt to the curved side or wrt the flat side? (Implementation wise I think it will always be considered on the flat side, since once you're within the margin of error it will decide that the plane contains() the point.)

[the remark this answers is not in the document]
Comment (abrahammurciano) [reply src/test/geometries/CylinderTests.java:16]: 1. There's no need to enumerate your questions if there's only one.
2. In my original comment I already said what you should do.
3. You should check both with an OR

**Review by abrahammurciano — APPROVED** [review]
